### ISSUE TITLE
can't install elilxir, raise DialyzerTest exception

### ISSUE THREAD

**Issue by oslivan** [elixir-lang/elixir#5670]
### Below is my erlang version
> Erlang/OTP 20 [DEVELOPMENT] [erts-8.2.1] [source-e528e53] [64-bit] [smp:4:4] [ds:4:4:10] [async-> > threads:10] [hipe] [kernel-poll:false]
> 
> Eshell V8.2.1  (abort with ^G)

### when I execute `make clean test` command, would raise some info below:
```bash
Kernel.DialyzerTest: failure on setup_all callback, test invalidated
match (=) failed
code:  [] = elixirc("#{fixture_path("dialyzer")} -o #{dir}")
right: 'warning: key \'\'__struct__\'\' will be overridden in expression\n  test/elixir/fixtures/dialyzer/raise.ex:5\n\nwarning: key \'\'__struct__\'\' will be overridden in expression\n  test/elixir/fixtures/dialyzer/protocol_opaque.ex:16\n\n'
     stacktrace:
       test/elixir/kernel/dialyzer_test.exs:31: Kernel.DialyzerTest.__ex_unit_setup_all_0/1
       test/elixir/kernel/dialyzer_test.exs:3: Kernel.DialyzerTest.__ex_unit__/2

       warning: key ''__struct__'' will be overridden in expression
       test/elixir/fixtures/parallel_struct/bar.ex:2
Finished in 21.5 seconds (11.9s on load, 9.6s on tests)
3453 tests, 1 failure, 7 invalid

Randomized with seed 969234
make: *** [test_stdlib] Error 1
````
Is my erlang's version is too higher? Thanks.

**Comment by josevalim** [comment]
You are using a development version of Erlang. There is no guarantee Elixir test suite will pass without warnings and similar on future Erlang versions. So you either need to use a released Erlang version, such as Erlang 19.2, or you use latest Erlang even if some Elixir tests will fail.

**Comment by oslivan** [comment]
@josevalim Thanks for you reply, now it was solved.
